src/base/kaldi-error.cc

```diff
@@ -18,156 +19,122 @@
-  ans->append(to_append)
-#endif  // defined(HAVE_CXXABI_H) && defined(HAVE_EXECINFO_H)
 
-#ifdef HAVE_EXECINFO_H
-std::string KaldiGetStackTrace() {
-#define KALDI_MAX_TRACE_SIZE 50
-#define KALDI_MAX_TRACE_PRINT 20  // must be even.
+
+static std::string KaldiGetStackTrace() {
   std::string ans;
-  void *array[KALDI_MAX_TRACE_SIZE];
-  size_t size = backtrace(array, KALDI_MAX_TRACE_SIZE);
-  char **strings = backtrace_symbols(array, size);
+#ifdef HAVE_EXECINFO_H
+  #define KALDI_MAX_TRACE_SIZE 50
+  #define KALDI_MAX_TRACE_PRINT 20  // must be even.
+  // buffer for the trace,
+  void *trace[KALDI_MAX_TRACE_SIZE];
+  // get the trace,
+  size_t size = backtrace(trace, KALDI_MAX_TRACE_SIZE);
+  // get the trace symbols,
+  char **trace_symbol = backtrace_symbols(trace, size);
+
+  // Compose the 'string',
+  ans += "[ Stack-Trace: ]\n";
   if (size <= KALDI_MAX_TRACE_PRINT) {
     for (size_t i = 0; i < size; i++) {
-      KALDI_APPEND_POSSIBLY_DEMANGLED_STRING(strings[i], &ans);
-      ans += "\n";
+      ans += Demangle(trace_symbol[i]) + "\n";
     }
   } else {  // print out first+last (e.g.) 5.
     for (size_t i = 0; i < KALDI_MAX_TRACE_PRINT/2; i++) {
-      KALDI_APPEND_POSSIBLY_DEMANGLED_STRING(strings[i], &ans);
-      ans += "\n";
+      ans += Demangle(trace_symbol[i]) + "\n";
     }
     ans += ".\n.\n.\n";
     for (size_t i = size - KALDI_MAX_TRACE_PRINT/2; i < size; i++) {
-      KALDI_APPEND_POSSIBLY_DEMANGLED_STRING(strings[i], &ans);
-      ans += "\n";
+      ans += Demangle(trace_symbol[i]) + "\n";
     }
     if (size == KALDI_MAX_TRACE_SIZE)
       ans += ".\n.\n.\n";  // stack was too long, probably a bug.
   }
-  free(strings);  // it's all in one big malloc()ed block.
-
 
-#ifdef HAVE_CXXABI_H  // demangle the name, if possible.
-#endif  // HAVE_CXXABI_H
+  // cleanup,
+  free(trace_symbol);  // it's okay, just the pointers, not the strings.
+#endif  // HAVE_EXECINFO_H
   return ans;
 }
-#endif
 
-void KaldiAssertFailure_(const char *func, const char *file,
-                         int32 line, const char *cond_str) {
-  MessageLogger ml(LogMessageEnvelope::Error, func, file, line);
-  ml.stream() << "Assertion failed: " << cond_str;
-#ifdef HAVE_EXECINFO_H
-  ml.stream() << "\nStack trace is:\n" << KaldiGetStackTrace();
-#endif
-}
 
-LogHandler SetLogHandler(LogHandler new_handler) {
-  LogHandler old_handler = g_log_handler;
-  g_log_handler = new_handler;
-  return old_handler;
-}
+/***** KALDI LOGIGNG *****/
 
-static void SendToLog(const LogMessageEnvelope &envelope,
-                      const char *message) {
-  // Send to a logging handler if provided.
-  if (g_log_handler != NULL) {
-    g_log_handler(envelope, message);
-    return;
+// Given a filename like "/a/b/c/d/e/f.cc",  GetShortFileName
+// returns "e/f.cc".  Does not currently work if backslash is
+// the filename separator.
+static const char *GetShortFileName(const char *filename) {
```

Comment:
> In src/base/kaldi-error.cc
> <https://github.com/kaldi-asr/kaldi/pull/755#discussion_r62720975>:
>
> >
> > -static void SendToLog(const LogMessageEnvelope &envelope,
> > -                      const char *message) {
> > -  // Send to a logging handler if provided.
> > -  if (g_log_handler != NULL) {
> > -    g_log_handler(envelope, message);
> > -    return;
> > +// Given a filename like "/a/b/c/d/e/f.cc",  GetShortFileName
> > +// returns "e/f.cc".  Does not currently work if backslash is
> > +// the filename separator.
> > +static const char *GetShortFileName(const char *filename) {
>
> There is no change in that function, it is closely associated to the
> constructor MessageLogger::MessageLogger(), which is the only place where
> it is used.
> For that reason I moved it to be right in front of it.
>
> There is no change in 'GetShortFileName' function otherwise. Just the
> 'narrative'
> of the '.cc' file is more logical.
>

Sure, but please don't move it because it messes up the history.  Others
who use Kaldi downstream may have made changes to that file, and any change
like this cause merge  problems.